R/create_base_vectors.R

```diff
@@ -6,294 +6,416 @@
+    dplyr::filter(age_class == 3)
+
+  sf::st_write(vri_class3, fs::path(out_dir, "vri_class3.gpkg"), append = FALSE)
+
+  #
+  # ######### this section is currently in test phase (related to removal fo deciduous areas in Date Creek after these were found to be highly selected and challanging to sample)
+  #
+  # # vri - deciduous leading - this needs to be appled for some
+  #
+  # # STILL TO DO - for areas with deciduous leading (AT, EP) Aspen and paper burch these should be seperated using the code "SPECIES_CD_1" == AT|EP.
+  # # ie important in Date Creek
+  #
+  vri_decid <- vri |>
+    dplyr::filter(SPECIES_CD_1 %in% c("AT", "EP")) # note might need to adjust for some areas of interest
+  sf::st_write(vri_decid, fs::path(out_dir, "vri_decid.gpkg"), append = FALSE)
+}
+
+
+### 3) Get harvest history and FTEN --------------------------------
+get_harvest <- function(aoi, out_dir) {
+  # 3) Download recent cutblocks (within last 20 years)
+  #message("\rDownloading cutblock layers")
+  # Uses date filter which filters cutblock ages less than 20 years, or 7305 days
+  cutblocks <- bcdata::bcdc_query_geodata("b1b647a6-f271-42e0-9cd0-89ec24bce9f7") |>
+    bcdata::filter(bcdata::INTERSECTS(aoi)) |>
+    bcdata::select(c("HARVEST_YEAR", "AREA_HA")) |>
+    bcdata::collect()
+
+  cutblocks <- cutblocks %>%
+    {
+      if (nrow(.) > 0) sf::st_intersection(., aoi) else .
+    } %>%
+    dplyr::filter(as.numeric(format(Sys.time(), "%Y")) - HARVEST_YEAR <= 20)
+
+  sf::st_write(cutblocks, fs::path(out_dir, "cutblocks.gpkg"), append = FALSE)
+
+  # 4) ften  - Download latest harvest layer
+  #message("\rDownloading ften harvest layers")
+
+  ften <- bcdata::bcdc_query_geodata("cff7b8f7-6897-444f-8c53-4bb93c7e9f8b") %>%
+    bcdata::filter(bcdata::INTERSECTS(aoi)) %>%
+    bcdata::select("HARVEST_AUTH_STATUS_CODE", "ISSUE_DATE", "CURRENT_EXPIRY_DATE_CALC", "LIFE_CYCLE_STATUS_CODE", "FILE_STATUS_CODE") |>
+    bcdata::collect() |>
+    dplyr::select(
+      "HARVEST_AUTH_STATUS_CODE", "ISSUE_DATE", "CURRENT_EXPIRY_DATE_CALC",
+      "LIFE_CYCLE_STATUS_CODE", "FILE_STATUS_CODE"
+    ) |>
+    dplyr::filter(ISSUE_DATE > 2000) # might need to adjust this to dynamic
+
+  sf::st_write(ften, fs::path(out_dir, "ften.gpkg"), append = FALSE)
+
+
+  cutblocks_ften <- dplyr::bind_rows(cutblocks, ften)
+
+  sf::st_write(cutblocks_ften, fs::path(out_dir, "cutblocks_ften.gpkg"), append = FALSE)
+
+  cli::cat_line()
+  cli::cli_alert_success(
+    "harvest layers downloaded and to written to {.path {out_dir}}"
+  )
+
+}
+
+### 5) TEM -----------------------------
+get_TEM <- function(aoi, out_dir) {
+  message("\rDownloading TEM layers")
+
+  tem <- bcdata::bcdc_query_geodata("0a83163b-a62f-4ce6-a9a1-21c228b0c0a3") %>%
+    bcdata::filter(bcdata::INTERSECTS(aoi)) %>%
+    bcdata::collect() %>%
+    {
+      if (nrow(.) > 0) sf::st_intersection(., aoi) else .
+    }
+
+  if (nrow(tem) > 0) {
+    sf::st_write(tem, fs::path(out_dir, "tem.gpkg"), append = FALSE)
+  }
```

Comment:
```suggestion
    bcdata::collect()

  if (nrow(tem) > 0) {
    tem <- sf::st_intersection(tem, aoi)
    sf::st_write(tem, fs::path(out_dir, "tem.gpkg"), append = FALSE)
  }
```

Same suggestion as in `get_BEC()`, however one question - what happens when `nrow(tem) == 0`? Right now it returns nothing in that case, with no message or warning...